pkg/backingstore/backingstore.go

```diff
@@ -427,7 +429,28 @@ func RunDelete(cmd *cobra.Command, args []string) {
 	backStore.Namespace = options.Namespace
 	backStore.Spec = nbv1.BackingStoreSpec{}
 
-	if !util.KubeDelete(backStore) {
+	nbClient := system.GetNBClient()
+
+	continueDelete := true
+	poolinfo, err := nbClient.ReadPoolAPI(nb.ReadPoolParams{Name: backStore.Name})
+	if err != nil {
+		if rpcErr, isRPCErr := err.(*nb.RPCError); !isRPCErr || rpcErr.RPCCode != "NO_SUCH_POOL" {
+			log.Fatalf(`❌ Failed to read BackingStore info: %s`, err)
+			continueDelete = false
+		}
+	} else {
+		switch poolinfo.Undeletable {
```

Comment:
I think we need to fail in any case of `poolinfo.Undeletable != ""`

Reply:
also on deleting? are we sure it will always be permanent error? shouldn't we reconcile on that?

Reply:
failing on all undeletables as bucket_deleting usually suggests a real issue in the core